src/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2022 Qiskit on IQM developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=no-name-in-module,import-error,super-init-not-called,too-many-instance-attributes
+"""
+Qiskit fake backend for IQM quantum computers.
+"""
+
+from typing import Union
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import CZGate, Measure, RGate
+from qiskit.providers import JobV1
+from qiskit.transpiler import Target
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit_aer import AerSimulator
+from qiskit_aer.noise import NoiseModel
+from qiskit_aer.noise.errors import depolarizing_error, thermal_relaxation_error
+
+from qiskit_iqm.fake_backends.chip_sample import IQMChipSample
+from qiskit_iqm.iqm_backend import IQMBackend
+
+from .quantum_architectures import Adonis
+
+
+class IQMFakeBackend(IQMBackend):
+    """
+    A fake backend class for an IQM chip sample.
+
+    Args:
+        chip_sample: An instance of a :class:`IQMChipSample` describing the
+        characteristics of a specific chip sample.
```

Comment:
No need to repeat the type.
```suggestion
        chip_sample: Describes the characteristics of a specific chip sample.
```